From bd5c3bbdf00280efd3e0dd36963d2df4fb45a8fa Mon Sep 17 00:00:00 2001
From: Ricardo Branco <rbranco@suse.de>
Date: Wed, 4 Mar 2026 14:05:46 +0100
Subject: [PATCH] tests: Increase timeout for wait_for_runtime_status

Signed-off-by: Ricardo Branco <rbranco@suse.de>
---
 test/test_helper.bash | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/test_helper.bash b/test/test_helper.bash
index 3ea2ee21..eae1a8e7 100644
--- a/test/test_helper.bash
+++ b/test/test_helper.bash
@@ -395,7 +395,7 @@ assert_stderr_contains() {
 wait_for_runtime_status() {
     local cid=$1
     local expected_status=$2
-    local how_long=5
+    local how_long=10
 
     t1=$(expr $SECONDS + $how_long)
     while [ $SECONDS -lt $t1 ]; do
