From 63c40feb8c9c276e0aa9f11b6c905c2c65168cc1 Mon Sep 17 00:00:00 2001
From: Ricardo Branco <rbranco@suse.de>
Date: Wed, 24 Sep 2025 18:30:40 +0200
Subject: [PATCH] test: Fix test race in 030-run

Signed-off-by: Ricardo Branco <rbranco@suse.de>
---
 test/system/030-run.bats | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/system/030-run.bats b/test/system/030-run.bats
index 297191d672..5c4fdc44c4 100644
--- a/test/system/030-run.bats
+++ b/test/system/030-run.bats
@@ -78,7 +78,7 @@ echo $rand        |   0 | $rand
     echo "$content" > $PODMAN_TMPDIR/tempfile
 
     run_podman run --rm -i --preserve-fds=2 $IMAGE sh -c "cat <&4" 4<$PODMAN_TMPDIR/tempfile
-    is "$output" "$content" "container read input from fd 4"
+    assert "$output" =~ "$content" "container read input from fd 4"
 }
 
 # 'run --preserve-fd' passes a list of additional file descriptors into the container
