From 14cb9d2a6cca996204b82824c4df2b27ec2ea0c3 Mon Sep 17 00:00:00 2001
From: Ricardo Branco <rbranco@suse.de>
Date: Fri, 7 Nov 2025 09:33:59 +0100
Subject: [PATCH] test/system: Update test to handle new error message from
 runc 1.3.3

Signed-off-by: Ricardo Branco <rbranco@suse.de>
---
 test/system/410-selinux.bats | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/test/system/410-selinux.bats b/test/system/410-selinux.bats
index f360e91446a..0c70379b9e3 100644
--- a/test/system/410-selinux.bats
+++ b/test/system/410-selinux.bats
@@ -259,7 +259,10 @@ function check_label() {
         # https://github.com/opencontainers/selinux/pull/148/commits/a5dc47f74c56922d58ead05d1fdcc5f7f52d5f4e
         #   from failed to set /proc/self/attr/keycreate on procfs
         #   to   write /proc/self/attr/keycreate: invalid argument
-        runc) expect=".*: \(failed to set\|write\) /proc/self/attr/keycreate.*" ;;
+        # runc 1.3.3 (temporarily?) changed the error message because of the fix to CVE-2025-52881, see
+        # https://github.com/opencontainers/runc/commit/2c5356e73f15b246729d03198bfb2c6c33454099
+        #   to   write fsmount:fscontext:proc/self/attr/keycreate: invalid argument
+        runc) expect=".*: \(failed\|write\).*proc/self/attr/keycreate.*" ;;
         *)    skip "Unknown runtime '$runtime'";;
     esac
 
